Sources/StreamChatSwiftUI/ChatChannel/ChatChannelView.swift

```diff
@@ -59,10 +59,13 @@ public struct ChatChannelView<Factory: ViewFactory>: View {
                 .if(viewModel.reactionsShown, transform: { view in
                     view.navigationBarHidden(true)
                 })
-                .if(!viewModel.reactionsShown && !viewModel.isMessageThread) { view in
+                .if(viewModel.channelHeaderType == .regular) { view in
                     view.modifier(factory.makeChannelHeaderViewModifier(for: viewModel.channel))
                 }
-                .if(!viewModel.reactionsShown && viewModel.isMessageThread) { view in
+                .if(viewModel.channelHeaderType == .typingIndicator) { view in
+                    view.modifier(factory.makeChannelHeaderViewModifier(for: viewModel.channel))
+                }
+                .if(viewModel.channelHeaderType == .messageThread) { view in
```

Comment:
This approach is way cleaner. The before used checks were very cryptic and I had no idea really what they meant. This makes things way more understandable. Enums rock!